### ISSUE TITLE
Various fleurons and Dingbats missing

### ISSUE THREAD

**Issue by Alexander-Wilms** [georgd/EB-Garamond#152]
I'm trying to [create a facsimile](https://github.com/Alexander-Wilms/Egenolff-Berner-specimen) of the Egenolff-Berner specimen and noticed that most of the ornamental Dingbats are missing in this typeface.

![image](https://user-images.githubusercontent.com/3226457/225975397-441301b9-243e-44b9-ad39-c73984d6461f.png)

![image](https://user-images.githubusercontent.com/3226457/225975311-e45719f6-00fe-4046-bfa1-180f617517ab.png)

![image](https://user-images.githubusercontent.com/3226457/225975519-492ed354-d061-4c53-9f7a-90d2b8169206.png)

![image](https://user-images.githubusercontent.com/3226457/225975572-419b6f6d-8ffb-4695-a7a2-0f36670c81bf.png)

![image](https://user-images.githubusercontent.com/3226457/225975629-cf23a835-4aff-4bd6-afa0-7bb0c1f1b192.png)

![image](https://user-images.githubusercontent.com/3226457/225975737-d513d589-42a2-4bae-b2d0-64e6b160898b.png) and a mirrored version, partially cut off in specimen

![image](https://user-images.githubusercontent.com/3226457/225976010-dad2f59a-71b6-46b3-bc62-f35d0a039086.png)

![image](https://user-images.githubusercontent.com/3226457/225976140-9b5c08f4-2a64-40ea-8454-0d94703141d3.png)

![image](https://user-images.githubusercontent.com/3226457/225976052-fae9f8f2-1558-462d-8352-85b99635e4b7.png)

The ornamental stamps at the bottom would be nice to have, as well:

![image](https://user-images.githubusercontent.com/3226457/225976535-a36c5963-6dd3-4baa-86b1-6dca29efbe58.png) etc.

**Comment by georgd** [comment]
Interesting project. I didn’t get as far as you in my try :)

I think, I found the stamps from the border in another revival font. Perhaps it was in the Fell Flowers font by Igino Marini: https://iginomarini.com/fell/the-revival-fonts/

**Comment by Alexander-Wilms** [comment]
Thanks for the info.

However, it seems like the Fell Flowers font only contains some of the stamp configurations seen in the EB specimen and the license does not allow modification.

Would these be in scope for EB Garamond? I could try to vectorize them.

**Comment by timgrei** [comment]
> Thanks for the info.
> 
> However, it seems like the Fell Flowers font only contains some of the stamp configurations seen in the EB specimen and the license does not allow modification.
> 
> Would these be in scope for EB Garamond? I could try to vectorize them.

I have also thought about vectorizing them. I think the Fell Flowers are just automatically traced, I can't understand, why you are not allowed to modify. In the books of Jean de Tournes there are thousands more of those ornaments. He seldom used one cut twice.

**Comment by timgrei** [comment]
Can one of you tell me, if these larger ones are wood cuts or lead stereotypes?

![grafik](https://user-images.githubusercontent.com/32984729/226064943-abda05be-ed85-400a-9c57-709946bee3d0.png)

**Comment by georgd** [comment]
I’d happily accept pull requests :) However, I’m still not sure about the target. Aldus leafs and variants thereof are definitely in the text fonts‘ scope. There are three unicode code points (floral heart) with various rotation variants:

- U+2619 ☙ REVERSED ROTATED FLORAL HEART BULLET (Miscellaneous Symbols)
- U+2766 ❦ FLORAL HEART (Dingbats)
- U+2767 ❧ ROTATED FLORAL HEART BULLET (Dingbats)

They are left pointing, downwards pointing and right pointing but no upwards pointing exists. The various leaf and achorn designs would be added as variants of these and should be made available via cv90 which already contains an alternative design.

**Comment by georgd** [comment]
> Can one of you tell me, if these larger ones are wood cuts or lead stereotypes?
> 
> ![grafik](https://user-images.githubusercontent.com/32984729/226064943-abda05be-ed85-400a-9c57-709946bee3d0.png)

I couldn’t tell. But regarding the size, I’d rather tend to expect a woodcut.

**Comment by timgrei** [comment]
> I couldn’t tell. But regarding the size, I’d rather tend to expect a woodcut.

That's what I thought, considering their quantity.
 
> They are left pointing, downwards pointing and right pointing but no upwards pointing exists. The various leaf and achorn designs would be added as variants of these and should be made available via cv90 which already contains an alternative design.

Wouldn't it be easier for developer and user to have a separate font for ornaments?

**Comment by timgrei** [comment]
I was asking because I thought that's where we could draw the line. Lead ornaments as part of the regular font and wood cuts in another.

**Comment by georgd** [comment]
> I’d happily accept pull requests :) However, I’m still not sure about the target. Aldus leafs and variants thereof are definitely in the text fonts‘ scope. There are three unicode code points (floral heart) with various rotation variants:
> 
> * U+2619 ☙ REVERSED ROTATED FLORAL HEART BULLET (Miscellaneous Symbols)
> * U+2766 ❦ FLORAL HEART (Dingbats)
> * U+2767 ❧ ROTATED FLORAL HEART BULLET (Dingbats)
> 
> They are left pointing, downwards pointing and right pointing but no upwards pointing exists. The various leaf and achorn designs would be added as variants of these and should be made available via cv90 which already contains an alternative design.

It’s similar for the index symbols. There are at least code points for the four principal directions (U+261C, U+261D, U+261E, U+261F).

The border designs, however, I don’t know where to put. I’ve always been considering a separate font for these. What do you think?

**Comment by georgd** [comment]
> I was asking because I thought that's where we could draw the line. Lead ornaments as part of the regular font and wood cuts in another.

Our last three comments were written simultaneously. We seem to think in the same direction, just setting the conditions on different sides of the history :). As I can not definitely tell which is a woodcut and which not, I’d rather take the unicode encoding as the measure for digitisation. If there’s a logical point in the encoding, it can be put there. If not, it should go in a separate font.

**Comment by timgrei** [comment]
That makes sense. So everything that is a Unicode symbol goes in the text font, everything else in _EB Garamond Zierrat_ (I love that word). (Maybe the former could just be in both?)

We could categorize the ornaments (tailpiece, frames, &c.) and assign to lowercase, uppercase and so on.

**Comment by timgrei** [comment]
> > I was asking because I thought that's where we could draw the line. Lead ornaments as part of the regular font and wood cuts in another.
> 
> Our last three comments were written simultaneously. We seem to think in the same direction, just setting the conditions on different sides of the history :). As I can not definitely tell which is a woodcut and which not, I’d rather take the unicode encoding as the measure for digitisation. If there’s a logical point in the encoding, it can be put there. If not, it should go in a separate font.

Last five* :D

**Comment by georgd** [comment]
> That makes sense. So everything that is a Unicode symbol goes in the text font, everything else in _EB Garamond Zierrat_ (I love that word). (Maybe the former could just be in both?)
> 
> We could categorize the ornaments (tailpiece, frames, &c.) and assign to lowercase, uppercase and so on.

EB Garamond Zierrat is nice, indeed :) It would be a classical ornamental font. The question would be if we‘d adhere to any encoding or template (another very popular font which could serve as such) or just do our own thing.

**Comment by timgrei** [comment]
Maybe my brain is weird, but when I think about that word I go: Zierrat –> Unrat –> Unkraut –> Blumen –> Ornamente. So it kind of has a recursive meaning to me.

We could look at those templates available and see if we agree with them. In most free ornament fonts it's all very messy.

**Comment by georgd** [comment]
Yes, it’s all very messy. I think, the only ones we should care abput would be default fonts in Windows/Mac/Linux. All the others we can safely disregard.

**Comment by timgrei** [comment]
What about the big foundries, Adobe, Monotype, ...

**Comment by georgd** [comment]
Do they follow a schema? I actually don’t have access to a relevant font by them.

**Comment by timgrei** [comment]
> EB Garamond Zierrat is nice, indeed :)

Maybe _Zierat_ would be more appropriate, which seems to have been the only form prior to the reform and has a nicer _Wortbild_. The _Grimmsches Wörterbuch_ even has the beautiful word _verzieraten_.

**Comment by georgd** [comment]
No, I think you can do it in the main branch. The real major retailors of fonts are the major operation system providers (someone might hate me for this). If there‘s no pattern to be found for an encoding of a not really special purpose font then we can assume that our prospects won't be covered. We’ll have to come up with our own encoding, then. For the users, we’ll have to document it, of course.

I’d suggest, however, to use the PUA for the ornaments and populate the ASCII block with the EBG ASCII glyphs as Icon fonts like Fontawesome do.

**Comment by timgrei** [comment]
> No, I think you can do it in the main branch. The real major retailors of fonts are the major operation system providers (someone might hate me for this). If there‘s no pattern to be found for an encoding of a not really special purpose font then we can assume that our prospects won't be covered. We’ll have to come up with our own encoding, then. For the users, we’ll have to document it, of course.

All right.


> I’d suggest, however, to use the PUA for the ornaments and populate the ASCII block with the EBG ASCII glyphs as Icon fonts like Fontawesome do.

Sorry, I'm not quite following. So the ASCII block has the same glyphs as the text font or put matching icons there?

**Comment by georgd** [comment]
> > EB Garamond Zierrat is nice, indeed :)
> 
> Maybe _Zierat_ would be more appropriate, which seems to have been the only form prior to the reform and has a nicer _Wortbild_. The _Grimmsches Wörterbuch_ even has the beautiful word _verzieraten_.

Linguistically it doesn‘t matter. The German ear doesn‘t hear a difference between a single or duplicate `r` in a German word in that place 😂

**Comment by timgrei** [comment]
I have German ears (also would be weird if an orthography reform would change pronunciation) :D But the word feels more like _heraus_, &c. (which we still write with one) than _Schiff(f)ahrt_. And I think _Zierrat_ looks weird.

**Comment by georgd** [comment]
> > I’d suggest, however, to use the PUA for the ornaments and populate the ASCII block with the EBG ASCII glyphs as Icon fonts like Fontawesome do.
> 
> Sorry, I'm not quite following. So the ASCII block has the same glyphs as the text font or put matching icons there?

Yes, the ASCII block would be present and show the same glyphs as EBG12. The actual symbols would be encoded in the PUA. Fontaweome starts them after F002.

**Comment by timgrei** [comment]
> > > I’d suggest, however, to use the PUA for the ornaments and populate the ASCII block with the EBG ASCII glyphs as Icon fonts like Fontawesome do.
> > 
> > 
> > Sorry, I'm not quite following. So the ASCII block has the same glyphs as the text font or put matching icons there?
> 
> Yes, the ASCII block would be present and show the same glyphs as EBG12. The actual symbols would be encoded in the PUA. Fontaweome starts them after F002.

Sounds good.

I always thought, it would be cool it those icon fonts would work with ligatures. So when you type `github` it becomes the github logo.

**Comment by georgd** [comment]
> I have German ears (also would be weird if an orthography reform would change pronunciation) :D But the word feels more like _heraus_, &c. (which we still write with one) than _Schiff(f)ahrt_. And I think _Zierrat_ looks weird.

German phonology doesn‘t have a different `r`. So whatever is codified by orthographic rules will just be an [R]. In the case of Zierrat it just reflects a supposed composition of zier + rat because the new orthography wants to be more logical in certain ethymological contexts. (I’m fine with this  as I was fine with the former because either is a modern interpretation...)

**Comment by georgd** [comment]
> > I always thought, it would be cool it those icon fonts would work with ligatures. So when you type `github` it becomes the github logo.

This can be done and is done by various fonts. but it depends on the user always being aware of what they’re being doing.

**Comment by timgrei** [comment]
> > I have German ears (also would be weird if an orthography reform would change pronunciation) :D But the word feels more like _heraus_, &c. (which we still write with one) than _Schiff(f)ahrt_. And I think _Zierrat_ looks weird.
> 
> German phonology doesn‘t have a different `r`. So whatever is codified by orthographic rules will just be an [R]. In the case of Zierrat it just reflects a supposed composition of zier + rat because the new orthography wants to be more logical in certain ethymological contexts. (I’m fine with this as I was fine with the former because either is a modern interpretation...)

I know (I never claimed the two forms would sound differently). I'm just saying a) it looks better and b) this is an old typeface and _Zierat_ is an old word used by typesetters in the olden times and unheard of by modern German speakers without this special interest. So I think _Zierat_ would be more appropriate. Sorry, for making us digress so much :D

Also, as a northern German I can't even make an [R] 🤣

**Comment by timgrei** [comment]
> 😕 

Egal :D

**Comment by georgd** [comment]
I wrote [R] to nor predetermine any of the possible realisations of /r/ that you’ll find between the Danish border and wherever south you find German speakers :D

**Comment by timgrei** [comment]
> wherever south you find German speakers

That would be Neuschwabenland.

**Comment by timgrei** [comment]
The books of de Turnes sometimes show this symbol: 
![grafik](https://user-images.githubusercontent.com/32984729/226077248-99021df2-2e67-4cc6-b249-0100d480fdbd.png)

Do you know what that is?

**Comment by Alexander-Wilms** [comment]
Maybe https://en.wikipedia.org/wiki/Medical_prescription ? Or Rex?

Cf. Domesday Book, Bedforscire: 
![image](https://user-images.githubusercontent.com/3226457/226077671-ee3a986a-03e8-4d3c-b095-a80fb9bf5d0d.png)

**Comment by timgrei** [comment]
Yes, it's from a book about cancer treatment, so the context makes sense. I have never seen this before, as this isn't on German prescriptions, so I didn't know how to google it and thought it was a symbol from that time.

**Comment by georgd** [comment]
Looks like the prescription symbol. EBG has it :)

**Comment by timgrei** [comment]
If we only use the PUA, you can only use them by copy-paste or some sort of library for TeX, HTML, etc., right? Wouldn't it then be irrelevant, how we sort them?

**Comment by georgd** [comment]
No, you can directly address them by using the code points like `\char"F002` (if this is still the correct syntax in LaTeX) or `&#xF002`. The sorting is indeed irrelevant :)

**Comment by timgrei** [comment]
Do you want us to document the source for each ornament? Because that would be a lot of work, that's maybe not necessary.

**Comment by georgd** [comment]
I don’t think it‘s necessary to document each ornament <> source relation but a list of all the sources might be interesting :)

**Comment by Alexander-Wilms** [comment]
Just for info if someone else is looking for these ornaments, there's also a CTAN package of IM Fell Flowers under the same proprietary license: https://ctan.org/pkg/imfellflowers

**Comment by timgrei** [comment]
I'm starting with [De peste commentarius](https://archive.org/embed/hin-wel-all-00002961-001) by de Tournes.

**Comment by timgrei** [comment]
Euclid's Geometry by Oliver Byrne uses the ornaments from the specimen.

**Comment by Alexander-Wilms** [comment]
How do you create them? With Inkscape? Do you have any tips?

**Comment by timgrei** [comment]
Yes, Inkscape. First I put the image in an AI upscaler (I use upscaly), then I use Inkscape's tool to trace bitmaps and manually clean it up.

**Comment by timgrei** [comment]
Obviously, some images have to be unwraped before.

**Comment by timgrei** [comment]
I thought, this was easier than the initials, but this took me two hours.

![grafik](https://user-images.githubusercontent.com/32984729/226216639-52c1cb11-c9be-4690-94d9-b006e63bb9bc.png)

**Comment by Alexander-Wilms** [comment]
I've started with "Granjon's six-piece combine flower on Great Primer A [FLC6] Units e-f" :

![image](https://user-images.githubusercontent.com/3226457/226475776-92cd0105-fd35-4317-85f1-5291bc62545c.png)
![Granjon_Flower_FLC_6_a_d_upscaled](https://user-images.githubusercontent.com/3226457/226475889-7f8c83da-862f-4d46-a6d9-fd6748a0c08b.svg)

**Comment by timgrei** [comment]
Okay, that's a book, we should get.

**Comment by Alexander-Wilms** [comment]
One thing to consider: Should we have one glyph per stamp and in addition to that combined ornaments?

Having combined ornaments makes it easier to typeset in a lot of cases, but the separate stamps are still needed in a few cases.

First try still using IM Fell:
![image](https://user-images.githubusercontent.com/3226457/226485764-272efae5-83b0-459d-9148-62c49da83401.png)
«The Fell Types are digitally reproduced by Igino Marini. www.iginomarini.com»

```tex
%!TEX program = lualatex
% adapted from https://tex.stackexchange.com/a/88093/38905
\PassOptionsToPackage{usenames,x11names,dvipsnames,svgnames}{xcolor}
\documentclass[]{scrartcl}
\usepackage{tikz}
\usetikzlibrary{calc,positioning}
\usepackage{geometry}
\geometry{a4paper,tmargin=1in,bmargin=1in,
	lmargin=1in,rmargin=1in,heightrounded}
%
\usepackage{fontspec}
\newfontfamily\granjonornaments{IMFeFlow1}
\newcommand\deco[3][black]{%
	\par\vspace{1ex}
	%\begin{center}
	\textcolor{#1}{%
		\fontsize{#2}{#2}\granjonornaments #3
	}
	%\end{center}
	\vspace*{1ex}\par
}
%
\newcommand{ \pattern}{EOPQ}
\begin{document}
	\begin{tikzpicture}[remember picture,overlay]
		\node[rotate=90,align=center] at ($(current page.west)+(.8cm,0.14cm)$){{\deco{15.1pt}{ \pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern}}};
		\node[rotate=90,align=center] at ($(current page.east)+(-.8cm,0.14cm)$){{\deco{15.1pt}{ \rotatebox[origin=c]{180}{\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern}}}};
		\node[rotate=0,align=center] at ($(current page.north)+(0.1cm,-.8cm)$){{\deco{14.35pt}{\rotatebox[origin=c]{0}
					{\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\hspace{2pt}}}}};
		\node[rotate=0,align=center] at ($(current page.south)+(0.1cm,.9cm)$){{\deco{14.35pt} {\rotatebox[origin=c]{180}{\hspace{2pt}\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern\pattern}}}};
		%\draw [rounded corners=8pt,line width=4pt,draw=DarkRed]($(current page.south west)+(.4cm,.5cm)$) rectangle ($(current page.north east)+(-.4cm,-.39cm)$);
	\end{tikzpicture}
	
\end{document}
```



Also, in a combined ornament one could have a bit of variation between each mirrored stamp.

**Comment by timgrei** [comment]
I would concentrate on the stamps themselves for now. Maybe that's something, we can think about, when we have finished a few ornaments.

Second one done.
![grafik](https://user-images.githubusercontent.com/32984729/226495315-bbebd3fe-1c9d-4164-b68a-d77223bbdaec.png)

**Comment by timgrei** [comment]
> Cf. the page preview [here](https://www.oakknoll.com/pages/books/127576/hendrik-d-l-vervliet/granjons-flowers-an-enquiry-into-granjons-giolitos-and-de-tournes-ornaments-1542-1586), which shows the ornament's name.

Maybe the author would provide us with a pdf, if we asked him. I mean, someone who writes such a book won't exactly be opposed to us digitalizing these as free software for the public to use.

**Comment by kess** [comment]
> Maybe the author would provide us with a pdf, if we asked him. I mean, someone who writes such a book won't exactly be opposed to us digitalizing these as free software for the public to use.

Sadly, he is no longer with us: https://en.wikipedia.org/wiki/H._D._L._Vervliet

**Comment by timgrei** [comment]
> > Maybe the author would provide us with a pdf, if we asked him. I mean, someone who writes such a book won't exactly be opposed to us digitalizing these as free software for the public to use.
> 
> Sadly, he is no longer with us: https://en.wikipedia.org/wiki/H._D._L._Vervliet

That's sad. Maybe he could have lifted the lambda mystery.

**Comment by Alexander-Wilms** [comment]
Into which .sfdir should these go?

* EBGaramond12-Regular.sfdir
* EBGaramond-Symbols.sfdir

**Comment by georgd** [comment]
I think it’s best to create a  separate EBGaramond-Ornaments.sfdir and map them to ASCII characters for now. I’m about to freeze the current EBG Text fonts and not to add any new features or glyphs there (except if there is an emergency). All further development should happen in the text fonts in the v2 branch. Later, adding them to the text fonts is fine. Then I’d say to put them in unencoded slots as alternatives of e.g. an aldus leaf (named uni2619.01, uni2619.02, uni2619.03 etc.) and make them available via OT features.
